runtime-light/stdlib/fork/fork-functions.h

```diff
@@ -27,7 +27,7 @@ inline constexpr auto DEFAULT_TIMEOUT_NS = std::chrono::duration_cast<std::chron
 // === Blocking API ================================================================================
 
 template<typename T>
-requires(is_optional<T>::value || std::same_as<T, mixed>) task_t<T> f$wait(int64_t fork_id, double timeout = -1.0) noexcept {
+requires(is_optional<T>::value || std::same_as<T, mixed> || is_class_instance<T>::value) task_t<T> f$wait(int64_t fork_id, double timeout = -1.0) noexcept {
```

Comment:
Please, add a test for that case. Also, you need to modify requires clause for the overload of that function.